fsharp-backend/src/BackendOnlyStdLib/HttpBytesClient.fs

```diff
@@ -0,0 +1,212 @@
+/// HTTP Client used by LibHttpBytesClient StdLib functions
+/// TODO: consider inlining this into LibHttpBytesClient
+module BackendOnlyStdLib.HttpBytesClient
+
+open System.Threading.Tasks
+open FSharp.Control.Tasks
+
+open System.IO
+open System.Net.Http
+
+open Prelude
+open LibExecution
+open LibExecution.RuntimeTypes
+open LibBackend
+open VendoredTablecloth
+
+module Telemetry = LibService.Telemetry
+
+type private HttpResult = { body : byte []; code : int; headers : HttpHeaders.T }
+type private ClientError = { error : string; code : int option }
+
+// There has been quite a history of .NET's HttpClient having problems,
+// including socket exhaustion and DNS results not expiring.
+// The history is outlined well here:
+// https://www.stevejgordon.co.uk/httpclient-connection-pooling-in-dotnet-core
+//
+// As of .NET 6 it seems we no longer need to worry about either socket
+// exhaustion or DNS issues. It appears that we can use either multiple HTTP
+// clients or just one, we use just one for efficiency.
+// See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0#alternatives-to-ihttpclientfactory
+//
+// Note that the number of sockets was verified manually, with:
+// `sudo netstat -apn | grep _WAIT`
+// TODO: I don't see where "the nunber of sockets" is actually configured?
+let private socketHandler : HttpMessageHandler =
+  new SocketsHttpHandler(
+    // Avoid DNS problems
+    PooledConnectionIdleTimeout = System.TimeSpan.FromMinutes 5.0,
+    PooledConnectionLifetime = System.TimeSpan.FromMinutes 10.0,
+
+    AutomaticDecompression = System.Net.DecompressionMethods.None,
```

Comment:
Should we do decompression? It was removed from the old version because we did it manually later, in order to keep the behavior identical. But we can't reasonably expect users to do their own decompression, especially since we barely provide the ability to do so. But maybe that's the goal?

Reply:
I'm happy either way, but don't have the experience to really know which way is appropriate. So, I"ll default to your judgment here, though intuitively I'm learning _slightly_ towards automatically decompressing.

Reply:
I'm currently of the stance of 'let's not automatically decompress, but avail some decompression functions.

I'm vaguely nervous about the implications here though:
Is there any reason to think that ASP.NET does something _fancy_ such that automatic .net httpclient -level decompression would be notably more efficient than doing so 'manually' via some function?
There will certainly be more `bytes` passed around - probably not a big deal?

Reply:
I added some commentary around this topic in-line.

Reply:
A thought I had, is that automatic decompression might not work for the specific initial  use case, as you probably don't want to decompress an asset being uploaded before passing it on to CDN.

On the other hand, you may be able to prevent decompression by passing headers which don't imply the body should be decompressed. But that feels like quite a dance, which is exactly what we're trying to avoid.

So I guess I'm leaning on the side of no decompression as well. The decompression dance is not trivial, so will need to be careful for that middleware.

It may be also that we can't get it as efficient as the asp.net decompression; that would be a good thing to learn, and try to address in the language and/or runtime. If we can't fix it we can always make a new version with improved performance, if it comes to that.